internal/controllers/aggregation/status.go

```diff
@@ -0,0 +1,85 @@
+package aggregation
+
+import (
+	"context"
+	"fmt"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	apiv1 "github.com/Azure/eno/api/v1"
+	"github.com/Azure/eno/internal/manager"
+	"github.com/go-logr/logr"
+)
+
+type statusController struct {
+	client client.Client
+}
+
+func NewStatusController(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&apiv1.Composition{}).
+		Owns(&apiv1.ResourceSlice{}).
+		WithLogConstructor(manager.NewLogConstructor(mgr, "statusAggregationController")).
+		Complete(&statusController{
+			client: mgr.GetClient(),
```

Comment:
This never needs to reach out to the overlay right? just aggregates from slices to compositions?

Reply:
Yeah exactly 👍